src/uport.js

```diff
@@ -35,28 +35,28 @@ class Uport {
    *
    * @memberof    Uport
    * @method      constructor
-   * @param       {String}            dappName                the name of your dapp
+   * @param       {String}            appName                the name of your app
    * @param       {Object}            opts                    optional parameters
    * @param       {String}            opts.rpcUrl             a JSON rpc url (defaults to https://ropsten.infura.io)
    * @param       {String}            opts.infuraApiKey       Infura API Key (register here http://infura.io/register.html)
    * @param       {Function}          opts.topicFactory       A function creating a topic
-   * @param       {Function}          opts.showHandler        Function to present QR code or other UX to approve request
-   * @param       {Function}          opts.mobileShowHandler  Function to request in mobile browsers
-   * @param       {Function}          opts.closeHandler       Function to hide UX created with showHandler after request is done
+   * @param       {Function}          opts.uriHandler        Function to present QR code or other UX to approve request
+   * @param       {Function}          opts.mobileUriHandler  Function to request in mobile browsers
+   * @param       {Function}          opts.closeUriHandler       Function to hide UX created with uriHandler after request is done
    * @return      {Object}            self
    */
 
   //  TODO do we need registry settings
-  constructor (dappName, opts = {}) {
-    this.dappName = dappName || 'uport-connect-app'
-    this.infuraApiKey = opts.infuraApiKey || this.dappName.replace(/\W/g, '')
+  constructor (appName, opts = {}) {
+    this.appName = appName || 'uport-connect-app'
+    this.infuraApiKey = opts.infuraApiKey || this.appName.replace(/\W+/g, '-')
 
-    this.rpcUrl = opts.rpcUrl || (INFURA_ROPSTEN + '/' + this.infuraApiKey)    
+    this.rpcUrl = opts.rpcUrl || (INFURA_ROPSTEN + '/' + this.infuraApiKey)
     this.isOnMobile = opts.isMobile || isMobile()
     this.topicFactory = opts.topicFactory || TopicFactory(this.isOnMobile)
-    this.showHandler = opts.showHandler || QRUtil.openQr
-    this.mobileShowHandler = opts.mobileShowHandler || mobileShowHandler
-    this.closeHandler = opts.closeHandler || QRUtil.closeQr
+    this.uriHandler = opts.uriHandler || openQr
+    this.mobileUriHandler = opts.mobileUriHandler || mobileUriHandler
+    this.closeUriHandler = opts.closeUriHandler || (this.uriHandler === openQr ? closeQr : undefined)
```

Comment:
I was having the same issues dealing with close, I think the problem here is if someone passed a uriHandler function name openQr then this would be true, since js is not/can't checking functional equivalence.